src/proposer.ts

```diff
@@ -455,6 +432,12 @@ async function publishBundle(data: string, signature: string, from: string) {
 		logger.error('Failed to save bundle/CID data:', error)
 		throw new Error('Database operation failed')
 	}
+
+	// Pin to Filecoin asynchronously (non-blocking)
+	const nonce = bundleData.nonce
+	pinBundleToFilecoin(cidToString, compressedData, nonce).catch((err) => {
```

Comment:
I made a note on Airtable and in the groupchat that we should ensure `compressedData` is deterministic, so different node operators at different times can reliably generate the same hash from the raw bundle data. `gzip` may inject metadata by default related to the time of compression that would make the compressed data different. Let's investigate.